src/extension.ts

```diff
@@ -86,25 +107,52 @@ const parseFile = async (source: string) => {
     return;
   }
 
-  const index = nodes.findIndex((node) => node.path === source);
+  const index = nodes.findIndex((node) => node.path === filePath);
   if (index !== -1) {
     nodes[index].label = title;
   } else {
-    nodes.push({ id: id(source), path: source, label: title });
+    nodes.push({ id: id(filePath), path: filePath, label: title });
   }
 
-  edges = edges.filter((edge) => edge.source !== id(source));
+  // remove edges based on an old version of this file
+  edges = edges.filter((edge) => edge.source !== filePath);
```

Comment:
Is it correct with `filePath` instead of `id(filePath)`?

Reply:
No :sweat_smile: Damn merges!